evap/staff/tests/test_views.py

```diff
@@ -904,40 +904,83 @@ def test_success_with_data(self):
+        cls.questionnaire_responsible = baker.make(Questionnaire, type=Questionnaire.Type.CONTRIBUTOR)
+        cls.evaluation_1 = baker.make(
+            Evaluation,
+            course=baker.make(Course, semester=semester, type=cls.course_type_1, responsibles=[cls.responsible]),
+        )
+        cls.evaluation_2 = baker.make(
+            Evaluation,
+            course=baker.make(Course, semester=semester, type=cls.course_type_2, responsibles=[cls.responsible]),
+        )
+        cls.evaluation_3 = baker.make(
+            Evaluation,
+            course=baker.make(Course, semester=semester, type=cls.course_type_3, responsibles=[cls.responsible]),
+        )
         baker.make(
             Contribution,
-            contributor=baker.make(UserProfile),
-            evaluation=iter([evaluation1, evaluation2]),
+            contributor=cls.responsible,
+            evaluation=cls.evaluation_1,
+            role=Contribution.Role.EDITOR,
+            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+        )
+        baker.make(
+            Contribution,
+            contributor=cls.responsible,
+            evaluation=cls.evaluation_2,
+            role=Contribution.Role.EDITOR,
+            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+        )
+        baker.make(
+            Contribution,
+            contributor=cls.responsible,
+            evaluation=cls.evaluation_3,
             role=Contribution.Role.EDITOR,
             textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-            _fill_optional=["contributor"],
-            _quantity=2,
-            _bulk_create=True,
         )
 
-    def test_assign_questionnaires(self):
-        page = self.app.get(self.url, user=self.manager)
-        assign_form = page.forms["questionnaire-assign-form"]
-        assign_form["Seminar"] = [self.questionnaire.pk]
-        assign_form["Lecture"] = [self.questionnaire.pk]
-        page = assign_form.submit().follow()
+    def test_questionnaire_assignment(self):
+        page = self.app.get(self.url, user=self.manager, status=200)
+        form = page.forms["questionnaire-assign-form"]
+        form[f"general-{self.course_type_1.id}"] = [self.questionnaire_1.pk, self.questionnaire_2.pk]
+        form[f"general-{self.course_type_2.id}"] = [self.questionnaire_2.pk]
+        form[f"contributor-{self.course_type_1.id}"] = [self.questionnaire_responsible.pk]
+        form["all-contributors"] = [self.questionnaire_contributor.pk]
 
-        for evaluation in self.semester.evaluations.all():
-            self.assertEqual(evaluation.general_contribution.questionnaires.count(), 1)
-            self.assertEqual(evaluation.general_contribution.questionnaires.get(), self.questionnaire)
+        response = form.submit().follow()
+        self.assertIn("Successfully", str(response))
+
+        self.assertEqual(
```

Comment:
I think you can shorten this by using `self.assertQuerySetEqual(a_queryset, [expected_object1, expected_obj2])`